hybrid/decomposers.py

```diff
@@ -725,31 +728,40 @@ def next(self, state, **runopts):
         return state.updated(subproblem=subbqm)
 
 
-def _all_minimal_covers(edgelist):
-    """Create all minimal covers, provided graphs contain fewer than 17 edges."""
+def _good_cover(edgelist, brute_force_threshold=16):
+    # Attempt to solve by brute force (tree decomposition)
+    tds = TreeDecompositionSolver()
+    bqm = dnx.algorithms.independent_set.maximum_weighted_independent_set_qubo(
+        G, lagrange=2.0) 
+    tree_width, elimination_order = min_fill_heuristic(bqm)
+    if tree_width <= tds.properties['max_treewidth']:
+        G = nx.from_edgelist(edgelist)
```

Comment:
`G` needs to be defined before it's used by dnx in line 734.
```suggestion
    G = nx.from_edgelist(edgelist)
    tds = TreeDecompositionSolver()
    bqm = dnx.algorithms.independent_set.maximum_weighted_independent_set_qubo(
        G, lagrange=2.0) 
    tree_width, elimination_order = min_fill_heuristic(bqm)
    if tree_width <= tds.properties['max_treewidth']:
```